feature-libs/asm/customer-360/components/asm-customer-promotion-listing/asm-customer-promotion-listing.component.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  Output,
+} from '@angular/core';
+import { GlobalMessageType } from '@spartacus/core';
+import { PromotionListEntry } from './asm-customer-promotion-listing.model';
+@Component({
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  selector: 'cx-asm-customer-promotion-listing',
+  templateUrl: './asm-customer-promotion-listing.component.html',
+})
+export class AsmCustomerPromotionListingComponent {
+  @Input() headerText: string;
+  @Input() emptyStateText: string;
+  @Input() applyButtonText: string;
+  @Input() applied: string;
+  @Input() removeButtonText: string;
+  @Input() entries: Array<PromotionListEntry> | null;
+  @Input() showAlert: boolean | null;
+  @Input() showAlertForApplyAction: boolean | null;
+  @Input() showRemoveButton: boolean;
+  @Input() showApplyButton: boolean;
+  @Input() isCustomerCoupon: boolean;
+  @Output() apply = new EventEmitter<PromotionListEntry>();
+  @Output() remove = new EventEmitter<PromotionListEntry>();
+  @Output() removeAlert = new EventEmitter();
+  @Output() removeAlertForApplyAction = new EventEmitter();
+  globalMessageType = GlobalMessageType;
+}
```

Comment:
[NITPICK - no need to change]
At the first sight such an API doesn't look ergonomic, each customization requires adding yet a new @Input. Did you consider using [ngTemplateOutlet](https://indepth.dev/posts/1405/ngtemplateoutlet) for customizations of this component?
No need to change, I'm just sharing knowledge about ngTemplateOutlet.